### DISCUSSION TITLE
Proper way to split a form into subcomponents

### DISCUSSION THREAD

**Discussion by emilpaw** [react-hook-form/react-hook-form#4954] [Q&A]
I spent half a day to figure out that it's not possible to move parts of a form to a seperate component like I did with `MyComponent` in this CSB: https://codesandbox.io/s/kind-williams-0653k?file=/src/App.js

It almost completely worked, the only issue it had was that it lost focus after the first character typed into the input. I figured out that this has probably something to do with the component re-rendering after the first typed character. The Form.Input which is not in a separate component does not do that and works like it should. But still I have to wonder, why does the Form.Input in the "root" component behaves differently from the one in `MyComponent`?

And what's a proper way to split a form into smaller components?

Thanks in advance.

**Comment by bluebill1049** [comment]
move your `MyComponent` outside of the main component, you are mount and unmount your inputs on each render. this should fix your problem.
